eth/reqresp.go

```diff
@@ -834,15 +963,14 @@ func (r *ReqResp) BlocksByRangeV2(ctx context.Context, pid peer.ID, firstSlot, l
 	if err != nil {
 		return blocks, fmt.Errorf("new %s stream to peer %s: %w", p2p.RPCBlocksByRangeTopicV2, pid, err)
 	}
-	defer stream.Close()
-	defer logDeferErr(stream.Reset, "failed closing stream") // no-op if closed
 
 	req := &pb.BeaconBlocksByRangeRequest{
 		StartSlot: primitives.Slot(firstSlot),
 		Count:     (lastSlot - firstSlot),
 		Step:      1,
 	}
 	if err := r.writeRequest(ctx, stream, req); err != nil {
+		stream.Reset()
```

Comment:
why only sometimes a reset and sometimes not even a close? (L999)

Reply:
Reset already closes both w/r streams and should be used when we spot an error. 
So either we Reset the stream on the error path, or we close the stream if everything goes as expected.
Did I miss any case where the stream is not properly closed?